snuba/migrations/runner.py

```diff
@@ -91,22 +92,37 @@ def run_migration(self, migration_key: MigrationKey) -> None:
     def _update_migration_status(
         self, migration_key: MigrationKey, status: Status
     ) -> None:
+        next_version = self._get_next_version(migration_key)
+
         statement = f"INSERT INTO {TABLE_NAME} FORMAT JSONEachRow"
         data = [
             {
                 "group": migration_key.group.value,
                 "migration_id": migration_key.migration_id,
                 "timestamp": datetime.now(),
                 "status": status.value,
-                # TODO: Version should be incremented each time we update that
-                # migration status
-                "version": 1,
+                "version": next_version,
             }
         ]
         self.__connection.execute(
             statement, data, settings={"load_balancing": "in_order"}
         )
 
+    def _get_next_version(self, migration_key: MigrationKey) -> int:
+        group = escape_string(migration_key.group.value)
+        migration_id = escape_string(migration_key.migration_id)
+        conditions = f"group = {group} AND migration_id = {migration_id}"
+
+        result = self.__connection.execute(
+            f"SELECT version FROM {TABLE_NAME} FINAL WHERE {conditions};",
+            settings={"load_balancing": "in_order"},
```

Comment:
It seems like this specific setting gets passed consistently for each query in here — isn't this what the client settings objects are for? Or is this special cased for some other reason?

Reply:
Good point, this probably applies anytime we are using the migrate settings, will move into there.